tests/suites/test_suite_psa_crypto.data

```diff
@@ -4397,6 +4397,10 @@ PSA verify hash: invalid algorithm for ECC key
 depends_on:PSA_WANT_ALG_RSA_PSS:PSA_WANT_ALG_SHA_256:PSA_WANT_KEY_TYPE_ECC_KEY_PAIR:MBEDTLS_PK_PARSE_C:PSA_WANT_ECC_SECP_R1_256:MBEDTLS_MD_C
 verify_hash_fail:PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_SECP_R1):"ab45435712649cb30bbddac49197eebf2740ffc7f874d9244c3460f54f322d3a":PSA_ALG_RSA_PSS(PSA_ALG_SHA_256):"":"":PSA_ERROR_INVALID_ARGUMENT
 
+PSA verify hash: ECDSA SECP256R1, valid key with invalid signature r=0,s=0
+depends_on:PSA_WANT_ALG_ECDSA:PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY:MBEDTLS_PK_PARSE_C:PSA_WANT_ECC_SECP_R1_256
+verify_hash_fail:PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_SECP_R1):"ab45435712649cb30bbddac49197eebf2740ffc7f874d9244c3460f54f322d3a":PSA_ALG_ECDSA_ANY:"ba7816bf8f01cfea414140de5dae2223b00361a396177a9cb410ff61f20015ad":"00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000":PSA_ERROR_INVALID_ARGUMENT
```

Comment:
That test data is incorrect: [`psa_verify_hash`](https://armmbed.github.io/mbed-crypto/html/api/ops/sign.html?highlight=psa_verify_hash#c.psa_verify_hash) is supposed to return [`PSA_ERROR_INVALID_SIGNATURE`](https://armmbed.github.io/mbed-crypto/html/api/library/status.html#c.PSA_ERROR_INVALID_SIGNATURE) if “`signature` is not the result of signing `hash` with algorithm `alg` using the private key corresponding to `key`”. The conditions for [`PSA_ERROR_INVALID_ARGUMENT`](https://armmbed.github.io/mbed-crypto/html/api/library/status.html#c.PSA_ERROR_INVALID_ARGUMENT) don't include a case for an invalid `signature`. Often the choice of error code is not a big deal, but `INVALID_SIGNATURE` vs others is, because typically the signature is untrusted and having it incorrect means you're getting garbage from an untrusted source, whereas `INVALID_ARGUMENT` indicates something wrong in trusted data (the public key) or in the application (using a bad key, a bad hash size, etc.).